File: dotnet/AIChatSDK/AIChatConfig.cs
namespace ChatFlow.SDK;

/// <summary>
/// Configuration options for the ChatFlow SDK client.
/// </summary>
public sealed class AIChatConfig
{
    /// <summary>
    /// Base URL of the ChatFlow API gateway. Defaults to http://localhost:5000.
    /// </summary>
    public string ApiUrl { get; set; } = "http://localhost:5000";

    /// <summary>
    /// Optional SignalR hub URL. Derived from ApiUrl when not set.
    /// </summary>
    public string? HubUrl { get; set; }

    /// <summary>
    /// API key for widget/server-side authentication.
    /// </summary>
    public string? ApiKey { get; set; }

    /// <summary>
    /// JWT access token for user authentication.
    /// </summary>
    public string? AccessToken { get; set; }

    /// <summary>
    /// Workspace ID to include in every request header.
    /// </summary>
    public string? WorkspaceId { get; set; }

    /// <summary>
    /// HTTP request timeout. Defaults to 30 seconds.
    /// </summary>
    public TimeSpan Timeout { get; set; } = TimeSpan.FromSeconds(30);

    /// <summary>
    /// Enable debug logging to the console.
    /// </summary>
    public bool Debug { get; set; }

    /// <summary>
    /// Derives the SignalR hub URL from ApiUrl if HubUrl is not explicitly set.
    /// </summary>
    internal string ResolvedHubUrl =>
        HubUrl ?? $"{ApiUrl.TrimEnd('/')}/hubs/chat";
}

File: swift/Sources/AIChatSDK/Config/AIChatConfig.swift
import Foundation

/// Configuration for AI Chat SDK
public struct AIChatConfig: Sendable {
    /// Base API URL
    public let apiURL: URL
    
    /// WebSocket URL (optional, will be derived from apiURL if not provided)
    public let wsURL: URL?
    
    /// API Key (optional, can use JWT instead)
    public let apiKey: String?
    
    /// Client ID for M2M authentication
    public let clientId: String?
    
    /// Client Secret for M2M authentication
    public let clientSecret: String?
    
    /// Request timeout
    public let timeout: TimeInterval
    
    /// Enable debug logging
    public let debug: Bool
    
    public init(
        apiURL: URL,
        wsURL: URL? = nil,
        apiKey: String? = nil,
        clientId: String? = nil,
        clientSecret: String? = nil,
        timeout: TimeInterval = 30,
        debug: Bool = false
    ) {
        self.apiURL = apiURL
        self.wsURL = wsURL
        self.apiKey = apiKey
        self.clientId = clientId
        self.clientSecret = clientSecret
        self.timeout = timeout
        self.debug = debug
    }
    
    /// Get WebSocket URL (derive from apiURL if not provided)
    public var websocketURL: URL {
        if let wsURL = wsURL {
            return wsURL
        }
        
        var components = URLComponents(url: apiURL, resolvingAgainstBaseURL: false)!
        components.scheme = apiURL.scheme == "https" ? "wss" : "ws"
        return components.url!
    }
}
